flink-core/src/main/java/org/apache/flink/configuration/NetworkEnvironmentOptions.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.configuration;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.annotation.docs.ConfigGroup;
+import org.apache.flink.annotation.docs.ConfigGroups;
+
+import static org.apache.flink.configuration.ConfigOptions.key;
+
+/**
+ * The set of configuration options relating to network stack.
+ */
+@PublicEvolving
+@ConfigGroups(groups = @ConfigGroup(name = "NetworkEnvironment", keyPrefix = "network.environment"))
```

Comment:
I'm reasonable certain that this doesn't work properly, and that you haven't regenerated the config docs through the generator.
Did you modify the `.html` files manually?

I'm asking since the name of the `.html` isn't correct; it is derived from the class names and config group names. For both this class and group it will be `network_environment_options.html`.

This ConfigGroup has a few issues too. `network.environment` as a `keyPrefix` at this time does not make sense since there is no matching option. Additionally, as described above the name clashes with the name of the class, which likely only works since there's no option actually matching the group.

Finally, it seems sensible to extract the netty options into a separate file to keep related options closer to each other. The current layout would be
```
<some generic network options (like taskmanager.network.memory)>
<netty options>
<some more generic network options LIke taskmanager.network.request-backoff)>
```

Reply:
Yes I modified the html file manually. Sorry for not fully understanding the config mechanism before. I would double check and adjust it.